diff -up Python-3.8.0/Makefile.pre.in.p501~ Python-3.8.0/Makefile.pre.in
--- Python-3.8.0/Makefile.pre.in.p501~	2019-11-03 02:49:37.579704405 +0100
+++ Python-3.8.0/Makefile.pre.in	2019-11-03 02:50:03.107713212 +0100
@@ -258,7 +258,8 @@ TCLTK_LIBS=	@TCLTK_LIBS@
 # To speed up profile generation, we don't run the full unit test suite
 # by default. The default is "-m test --pgo". To run more tests, use
 # PROFILE_TASK="-m test --pgo-extended"
-PROFILE_TASK=	@PROFILE_TASK@
+# PROFILE_TASK=	@PROFILE_TASK@
+PROFILE_TASK_PYBENCH=$(srcdir)/Tools/pybench/pybench.py -n 20
 
 # report files for gcov / lcov coverage report
 COVERAGE_INFO=	$(abs_builddir)/coverage.info
@@ -496,7 +497,7 @@ build_all_generate_profile:
 
 run_profile_task:
 	@ # FIXME: can't run for a cross build
-	$(LLVM_PROF_FILE) $(RUNSHARED) ./$(BUILDPYTHON) $(PROFILE_TASK) || true
+	$(LLVM_PROF_FILE) $(RUNSHARED) ./$(BUILDPYTHON) $(PROFILE_TASK_PYBENCH) || true
 
 build_all_merge_profile:
 	$(LLVM_PROF_MERGER)
