diff -up Python-3.13.3/Modules/posixmodule.c.4~ Python-3.13.3/Modules/posixmodule.c
--- Python-3.13.3/Modules/posixmodule.c.4~	2025-04-08 15:54:08.000000000 +0200
+++ Python-3.13.3/Modules/posixmodule.c	2025-04-25 23:19:10.932544794 +0200
@@ -12609,7 +12609,7 @@ static PyObject *
 os_major_impl(PyObject *module, dev_t device)
 /*[clinic end generated code: output=4071ffee17647891 input=b1a0a14ec9448229]*/
 {
-    return major_minor_conv(major(device));
+    return major_minor_conv(gnu_dev_major(device));
 }
 
 
@@ -12626,7 +12626,7 @@ static PyObject *
 os_minor_impl(PyObject *module, dev_t device)
 /*[clinic end generated code: output=306cb78e3bc5004f input=2f686e463682a9da]*/
 {
-    return major_minor_conv(minor(device));
+    return major_minor_conv(gnu_dev_minor(device));
 }
 
 
